graph/memstore/quadstore.go

```diff
@@ -117,17 +116,17 @@ func (qs *QuadStore) ApplyDeltas(deltas []graph.Delta, ignoreOpts graph.IgnoreOp
 		case graph.Add:
 			if !ignoreOpts.IgnoreDup {
 				if _, exists := qs.indexOf(d.Quad); exists {
-					return graph.ErrQuadExists
+					return &graph.QuadError{Action: d.Action, Quad: d.Quad, Err: graph.ErrQuadExists}
```

Comment:
1,2) Where was no references because nobody cared about checking this behavior for some reason. #378 does this. Maybe some apps also do. Unexporting sounds good to signal that something changed to the user, but it also means that backends could not use the error.
3) Looks good this way.